apps/demo-app/pages/api/check-signature.ts

```diff
@@ -0,0 +1,205 @@
+import type { NextApiRequest, NextApiResponse } from "next";
+import { verifyADR36Amino } from "@keplr-wallet/cosmos";
+
+// This import will need to change based on the chain you are confirming against.
+import { testnetChainInfo } from "@burnt-labs/constants";
+import { QueryGrantsResponse } from "cosmjs-types/cosmos/authz/v1beta1/query";
+
+function isString(test: unknown): test is string {
+  return typeof test === "string";
+}
+
+/**
+ * Verify that the given XION signature corresponds to the given message and address.
+ *
+ * @param address - The address that is supposed to have signed the message.
+ * @param pubKey - The public key associated with the session address base64 encoded.
+ * @param messageString - The message that is supposed to have been signed.
+ * @param signature - The signature to verify against the message and address.
+ * @returns True if the signature is valid, false otherwise.
+ */
+export function verifyXionSignature(
+  address: string,
+  pubKey: string,
+  messageString: string,
+  signature: string,
+): boolean {
+  const signatureBuffer = Buffer.from(signature, "base64");
+  const uint8Signature = new Uint8Array(signatureBuffer); // Convert the buffer to an Uint8Array
+  const pubKeyValueBuffer = Buffer.from(pubKey, "base64"); // Decode the base64 encoded value
+  const pubKeyUint8Array = new Uint8Array(pubKeyValueBuffer); // Convert the buffer to an Uint8Array
```

Comment:
Is there a situation where any of these parameters are empty/malformed/invalid? Do we need to do any checking inside the function before performing the changes?

Reply:
The verifyADR36Amino call should confirm these for us.